eda-env/Lib/site-packages/parso/grammar.py

```
@@ -0,0 +1,264 @@
+    except KeyError:
+        try:
+            with open(path) as f:
+                bnf_text = f.read()
+
+            grammar = PythonGrammar(version_info, bnf_text)
+            return _loaded_grammars.setdefault(path, grammar)
+        except FileNotFoundError:
+            message = "Python version %s.%s is currently not supported." % (
+                version_info.major, version_info.minor
+            )
+            raise NotImplementedError(message)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)